.github/workflows/ci_linux_arm64_clang.yml

```diff
@@ -27,39 +27,44 @@ jobs:
   setup:
     uses: ./.github/workflows/setup.yml
 
-  # TODO(#18238): migrate to new runner cluster
-  #   Or switch to GitHub-hosted arm64 runners when available?
   linux_arm64_clang:
     needs: setup
-    runs-on:
-      - self-hosted # must come first
-      - runner-group=${{ needs.setup.outputs.runner-group }}
-      - environment=${{ needs.setup.outputs.runner-env }}
-      - arm64
-      - os-family=Linux
+    # See https://gitlab.arm.com/tooling/gha-runner-docs
+    runs-on: ah-ubuntu_22_04-c7g_4x-50
+    container:
+      image: ghcr.io/iree-org/cpubuilder_ubuntu_jammy@sha256:f3e2c56680b6007f29c8e191d93dfcb4f8800802e3caeb4851da90f2a1a0b5a5
+    defaults:
+      run:
+        shell: bash
     env:
       BUILD_DIR: build-arm64
     steps:
       - name: "Checking out repository"
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
         with:
           submodules: true
-      - name: "Building IREE"
-        env:
-          IREE_WRITE_REMOTE_CCACHE: ${{ needs.setup.outputs.write-caches }}
+      - name: Install Python requirements
+        run: python3 -m pip install -r ./runtime/bindings/python/iree/runtime/build_requirements.txt
+      # TODO(#18557): enable sccache (watching for network bandwidth charges
+      #               between the cache server and AWS VMs)
+      - name: CMake - configure
         run: |
-          ./build_tools/github_actions/docker_run.sh \
```

Comment:
This is also the last active use of the `docker_run.sh` script and the GCP ccache storage bucket, so those can be removed if we merge this :D. With a bit more cleanup we can also remove the `IREE_BUILD_SETUP_PYTHON_VENV` environment variable and maybe even the whole `build_tools/cmake/build_all.sh` script